Sensors/SMCDellSensors/KeyImplementations.cpp

```diff
@@ -20,19 +20,15 @@ static inline UInt16 decode_fpe2(UInt16 value) {
 }
 
 SMC_RESULT F0Ac::readAccess() {
```

Comment:
Note that this translates to sequentially consistent access. IIRC `IOSimpleLock` provides a weaker acquire-release semantics.

Reply:
Could you point me out: where exactly should I use IOSimpleLock? Instead of IOLockLock? Only in this source file?

Reply:
My comment is emphasising the difference between `atomic_load/store(...)` and `IO(Simple)Lock`. The former implicitly uses sequentially consistent ordering, see https://en.cppreference.com/w/cpp/atomic/memory_order. The latter performs an acquire or release (`osfmk/kern/locks.c`). For performance reasons, you almost certainly want acquire/release, unless the code is functionally incorrect without `seq_cst`.

If you are wondering about the difference between `IOLock` and `IOSimpleLock`, you can check `IOKit/IOLocks.h`. Basically, `IOLock` is a heavyweight mutex that is meant for longer waiting periods, while `IOSimpleLock` disables preemption and is intended to be held for a short period. You can decide which lock is to be used by estimating the duration of each critical section.

Reply:
I see, thank you!